ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/SnapshotTrieLogManager.java

```diff
@@ -118,8 +143,11 @@ public TrieLogLayer getTrieLog() {
     }
 
     @Override
-    public MutableWorldState getMutableWorldState() {
-      return snapshot.get();
+    public synchronized BonsaiSnapshotWorldState getMutableWorldState() {
```

Comment:
is it easy to use an optional here ?

Reply:
I considered it, but it was a little messy - I think due to generics IIRC.  I will double check